packages/orchestrator/src/event.ts

```diff
@@ -0,0 +1,30 @@
+import * as grpc from 'grpc'
+import * as base58 from '@mesg/api/lib/util/base58'
+import { Client } from './client'
+import * as Type from './typedef/event'
+import * as API from './api/event'
+
+export type StreamRequest = {
+  filter: {
+    hash?: string
+    instanceHash?: string
+    key?: string
+  }
+}
+
+export default class Event extends Client {
+  constructor(endpoint: string) {
+    super(endpoint, 'Event')
+  }
+
+  public stream(request: StreamRequest, signature: string): grpc.ClientReadableStream<Type.mesg.types.IEvent> {
```

Comment:
`StreamRequest` and all attributes of these functions are here in order to "normalize" the data between the serialization in json/amino and the one in proto.
The signature is based on the json serialization while the payload is sent in proto which is really confusing.